tools/topology/topology2/cavs-nocodec-bt.conf

```diff
@@ -77,12 +80,21 @@ Object.Dai {
 		sample_bits		32
 		quirks			"lbm_mode"
 		io_clk		$MCLK
+		version         $SSP_BLOB_VERSION
+
 		Object.Base.hw_config.1 {
 			name	"SSP0"
 			id      0
 			mclk_freq       $MCLK
 			bclk_freq       3072000
 			tdm_slot_width  32
+			IncludeByKey.PLATFORM {
+				"mtl" {
+					Object.Base.link_config.1 {
+						clock_source	2
```

Comment:
can we not have some sort of enum that is self explanatory or a comment. This is very hard to review.

Reply:
I'd not pollute the hw_config class with NHLT-specific stuff at all. The clock_source is a SSP-specific value that can be set in ssp.conf based on PLATFORM definition

Reply:
what i'd think you need is something like this: 
```
diff --git a/tools/topology/topology2/include/dais/ssp.conf b/tools/topology/topology2/include/dais/ssp.conf
index 9c73e7562..99fc9024e 100644
--- a/tools/topology/topology2/include/dais/ssp.conf
+++ b/tools/topology/topology2/include/dais/ssp.conf
@@ -111,6 +111,25 @@ Class.Dai."SSP" {
        # platform clock frequency
        DefineAttribute.io_clk {}
 
+       DefineAttribute."clock_source" {
+               constraints {
+                       !valid_values [
+                               "XTAL"
+                               "CARDINAL"
+                               "PLL"
+                               "MCLK"
+                               "WOV_CRO"
+                       ]
+                       !tuple_values [
+                               0
+                               1
+                               2
+                               3
+                               4
+                       ]
+               }
+       }
+
        attributes {
                !constructor [
                        "name"
@@ -140,4 +159,10 @@ Class.Dai."SSP" {
        frame_pulse_width       0
        tdm_padding_per_slot    false
        version                 $SSP_BLOB_VERSION
+
+       IncludeByKey.PLATFORM {
+               "mtl"   {
+                       clock_source    PLL
+               }
+       }
 }
```

Reply:
> +       IncludeByKey.PLATFORM {
> +               "mtl"   {
> +                       clock_source    PLL

I hope we are using the 24.576MHz cardinal clock that comes for free and not the PLL? That index 2 seems completely wrong...

Reply:
@plbossart I couldnt be wrong with the indices, I was just giving an example. @juimonen please correct me

Reply:
all this is good and dandy, but it would require me to rewrite/move the alsa-utils auxdata parser

Reply:
We can debate about the sw-philosophical issues for a long time... So ssp blob is created per hw_config, like in BT we have 3 different hw_configs, we create 3 different blobs into nhlt. Aux data is embedded by the spec after the blob. So if the aux data is moved into ssp level, it would mean that all blobs would need a copy of the _same_ aux data. I understand that in the link clock case this could be fine, but link clock is not the only aux data item. So if someone can guarantee me that any aux data items will not be different in any hw_configs per ssp, the move to ssp level would be justified. Otherwise I see the hw_config quite logical place for it.

Reply:
I will repeat my suggestion to ALWAYS provide the AUX information with the clock source, and have the firmware discard what it cannot understand.

The point is that if you select a bitclock that divided from 24.576 MHz then you *have* to select the clock_source 1. If you derive the bitclock from 19.2 then clock_source 0 it is.

There's no differences between platforms, the mapping between clock sources is always the same, and we only have to select between 0 and 1 for now.

In other words, rather than keep this optional make it a mandatory one-size-fits-all.